aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/RunFaceLivenessSession.kt

```diff
@@ -78,12 +83,75 @@ internal class RunFaceLivenessSession(
+        if (sessionInformation.challenge.faceMovementAndLightChallenge != null) {
+            val challenge = sessionInformation.challenge.faceMovementAndLightChallenge
+
+            // Face target challenge
+            sessionChallenges.add(getFaceTargetChallenge(challenge.ovalParameters, challenge.challengeConfig))
+
+            // Freshness color challenge
+            val colorChallengeType = ColorChallengeType.SEQUENTIAL
+            val challengeColors = mutableListOf<ColorDisplayInformation>()
+            challenge.colorSequences.forEachIndexed { _, colorSequence ->
+                val currentColor = colorSequence.freshnessColor.rGB
+                val rgbColor = RgbColor(currentColor[0], currentColor[1], currentColor[2])
+                var duration = colorSequence.flatDisplayDuration
+                var shouldScroll = false
+                if (colorSequence.flatDisplayDuration == 0f) {
+                    duration = colorSequence.downscrollDuration
+                    shouldScroll = true
+                }
+                challengeColors.add(
+                    ColorDisplayInformation(
+                        rgbColor,
+                        duration,
+                        shouldScroll
+                    )
+                )
+            }
+            val colorChallenge = ColorChallenge(
+                livenessWebSocket.challengeId,
+                colorChallengeType,
+                challengeColors.toList()
+            )
+            sessionChallenges.add(colorChallenge)
+        } else if (sessionInformation.challenge.faceMovementChallenge != null) {
+            val challenge = sessionInformation.challenge.faceMovementChallenge
+
+            // Face target challenge
+            sessionChallenges.add(getFaceTargetChallenge(challenge.ovalParameters, challenge.challengeConfig))
+        }
+
+        return sessionChallenges.toList()
+    }
+
+    private fun getChallengeId(): String = livenessWebSocket.challengeId
+
+    private fun processChallengeType(): FaceLivenessChallengeType =
+        when (livenessWebSocket.challengeType) {
+            ChallengeType.FaceMovementChallenge -> FaceLivenessChallengeType.FaceMovementChallenge
+            ChallengeType.FaceMovementAndLightChallenge -> FaceLivenessChallengeType.FaceMovementAndLightChallenge
+            null -> {
+                // A ChallengeType event is not returned when the v1 version of FaceMovementAndLightChallenge
+                // is requested so we'll check here for backwards compat
+                if (clientSessionInformation.challengeVersions != null &&
+                    clientSessionInformation.challengeVersions!!.contains(
+                            Challenge(ChallengeType.FaceMovementAndLightChallenge, "1.0.0")
+                        )
+                ) {
+                    FaceLivenessChallengeType.FaceMovementAndLightChallenge
+                } else {
+                    FaceLivenessChallengeType.Unknown
```

Comment:
I don't think we need unknown. We should catch exception and provide exception in onError

Reply:
With the changes I'm making, `challengeType` should never be null by the time we get to this point. An exception will be provided earlier in the flow.